src/acceptance-test/java/org/zalando/nakadi/webservice/utils/TestStreamingClient.java

```diff
@@ -58,8 +63,12 @@ public void run() {
             final String url = format("{0}/subscriptions/{1}/events?{2}", baseUrl, subscriptionId, params);
             final HttpURLConnection conn = (HttpURLConnection) new URL(url).openConnection();
             token.ifPresent(token -> conn.setRequestProperty("Authorization", "Bearer " + token));
-            if (conn.getResponseCode() != HttpURLConnection.HTTP_OK) {
-                throw new IOException("Response code is " + conn.getResponseCode());
+            responseCode = conn.getResponseCode();
+            conn.getHeaderFields().entrySet().stream()
+                    .filter(entry -> entry.getKey() != null)
```

Comment:
Is it really the case when the header name is null?
Because if not - you can simply do: `headers.putAll(conn.getHeaderFields())`